src/CloudKit/CKCompat.cs

```diff
@@ -30,6 +35,94 @@ public CKNotificationID ()
 		{
 		}
 	}
+
+	public partial class CKContainer {
+
+		[iOS (8, 0), Mac (10, 10)]
+		[Obsolete ("Use 'DiscoverAllIdentities' instead. NotSupportedException'")]
+		public virtual void DiscoverAllContactUserInfos (Action<CKDiscoveredUserInfo[], NSError> completionHandler) 
+			=> throw new NotSupportedException ();
+
+		[iOS (8, 0), Mac (10, 10)]
+		[Obsolete ("Use 'DiscoverAllIdentities' instead. Always throw a 'NotSupportedException'")]
+		public virtual Task<CKDiscoveredUserInfo[]> DiscoverAllContactUserInfosAsync ()
+			=> Task.FromException<CKDiscoveredUserInfo[]> (new NotSupportedException ());
+
+		[iOS (8, 0), Mac (10, 10)]
+		[Obsolete ("Use 'DiscoverUserIdentityWithEmailAddress' instead. Always throw a 'NotSupportedException'")]
+		public virtual void DiscoverUserInfo (string email, Action<CKDiscoveredUserInfo, NSError> completionHandler)
+			=> throw new NotSupportedException ();
+
+		[iOS (8, 0), Mac (10, 10)]
+		[Obsolete ("Use 'DiscoverUserIdentityWithEmailAddress' instead. Always throw a 'NotSupportedException'")]
+		public virtual Task<CKDiscoveredUserInfo> DiscoverUserInfoAsync (string email)
+			=> Task.FromException<CKDiscoveredUserInfo> (new NotSupportedException ());
+
+		[iOS (8, 0), Mac (10, 10)]
+		[Obsolete ("Use 'DiscoverUserIdentity' instead. Always throw a 'NotSupportedException'")]
+		public virtual void DiscoverUserInfo (CKRecordID userRecordId, Action<CKDiscoveredUserInfo, NSError> completionHandler)
+			=> throw new NotSupportedException ();
+
+		[iOS (8, 0), Mac (10, 10)]
+		[Obsolete ("Use 'DiscoverUserIdentity' instead. Always throw a 'NotSupportedException'")]
+		public virtual Task<CKDiscoveredUserInfo> DiscoverUserInfoAsync (CKRecordID userRecordId)
+			=> Task.FromException<CKDiscoveredUserInfo> (new NotSupportedException ());
+	}
+
+	public partial class CKDiscoverAllContactsOperation {
+
+		[Obsolete ("Empty stub (not a public API).")]
+		public virtual Action<CKDiscoveredUserInfo[], NSError> DiscoverAllContactsHandler { get; set; }
+
+	}
+
+	[iOS (8,0), Mac (10,10)]
+	public delegate void CKDiscoverUserInfosCompletionHandler (NSDictionary emailsToUserInfos, NSDictionary userRecordIdsToUserInfos, NSError operationError);
+	
+	public partial class CKDiscoverUserInfosOperation { 
+
+		[iOS (8,0), Mac (10,10)]
+		[Obsolete ("Empty stub (not a public API).")]
+		public virtual CKDiscoverUserInfosCompletionHandler Completed { get; set; }
+	}
+
+	public partial class CKDiscoverUserInfosOperation {
+		[Obsolete ("Always throws NotSupportedException (). (not a public API).")]
```

Comment:
This is close to an existing string `"Always throw a 'NotSupportedException' (not a public API)."` and this pattern is unlike others.

Having different string is bad since they can't be shared inside the assembly, leading to larger size apps. The linker can remove some but it's not always enabled (e.g. interpreter). I'll file an issue on this.